packages/serve-runtime/src/useForwardHeaders.ts

```diff
@@ -7,20 +7,22 @@ export interface ForwardHeadersPluginOptions {
 export function useForwardHeaders(headerNames: string[]): MeshServePlugin {
   return {
     onFetch({ options, setOptions, context }) {
-      const forwardedHeaders: Record<string, string> = {};
-      for (const headerName of headerNames) {
-        const headerValue = context.headers[headerName];
-        if (headerValue) {
-          forwardedHeaders[headerName] = headerValue;
+      if (context?.request?.headers) {
```

Comment:
When is it possible that a request is not present in the context ?

Reply:
When fetch is called independently from the user call such as introspection of proxied API.